test/e2e/replica_set.go

```diff
@@ -94,6 +94,9 @@ func ReplicaSetServeImageOrFail(f *Framework, test string, image string) {
 	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
 
 	pods, err := podsCreated(f.Client, f.Namespace.Name, name, replicas)
+	if pods == nil || err != nil {
```

Comment:
I'd just check `err != nil` here; I think you can safely assume that if there's no error, `pods` isn't nil, and if it is, that should be fixed in `podsCreated` or properly handled separately.